File: Domain/src/Obilazak.cs
namespace Domain;

public class Obilazak
{
    public Guid ID { get; set; }
    public string Naziv { get; set; }
    public string Opis { get; set; }
    public DateTime DatumOdrzavanja { get; set; }
    public int BrojMaxPolaznika { get; set; }
    public string MestoOkupljanja { get; set; }
    public ICollection<Komentar> Komentari { get; set; } = new List<Komentar>();
    public string Alergije { get; set; }
    public Vodic Vodic { get; set; }
    public ICollection<PrijavljeniObilazak> Ucesnici { get; set; } = new List<PrijavljeniObilazak>();
    public Podrucje Podrucje { get; set; }
    public Ruta Ruta {get; set;}
}

File: client-app/src/App/modules/Obilazak.ts
import { Komentar } from "./Komentari"
import { Profil } from "./Profil"
import { Ruta } from "./Ruta";
import { Vodic } from "./Vodic";

export interface Obilazak {
    id: string;
    naziv: string;
    opis: string;
    datumOdrzavanja: Date | null;
    brojMaxPolaznika: number | null;
    mestoOkupljanja: string;
    komentari?: Komentar[];
    alergije?: any[];
    isCancelled?: boolean;
    vodic: Vodic;
    ide:boolean;
    isVodic:boolean;
    ucesnici: Profil[];
    ruta: Ruta;
}

export class Obilazak implements Obilazak{

    constructor(init?:ObilazakFromValues){
        Object.assign(this,init);
    }
}

export class ObilazakFromValues{
    id?: string = undefined;
    naziv: string = '';
    opis: string = '';
    datumOdrzavanja: Date | null = null;
    brojMaxPolaznika: number | null = null;
    mestoOkupljanja: string = '';
    isCancelled: boolean = false;

    constructor(tour?: ObilazakFromValues){
        if(tour){
            this.id = tour.id;
            this.naziv = tour.naziv;
            this.opis = tour.opis;
            this.datumOdrzavanja = tour.datumOdrzavanja;
            this.brojMaxPolaznika = tour.brojMaxPolaznika;
            this.mestoOkupljanja = tour.mestoOkupljanja;
        }
    }
}